src/main/java/edu/harvard/iq/dataverse/FileDownloadServiceBean.java

```diff
@@ -254,10 +260,6 @@ public void explore(GuestbookResponse guestbookResponse, FileMetadata fmd, Exter
                 dataFile = guestbookResponse.getDataFile();
             }
         }
-        //For tools to get the dataset and datasetversion ids, we need a full DataFile object (not a findCheapAndEasy() copy)
-        if(dataFile.getFileMetadata()==null) {
-            dataFile=datafileService.find(dataFile.getId());
-        }
```

Comment:
I'm not sure I understand why this was removed. We don't need it, I can only assume.

Reply:
Yes - I added it in the original change to send the dataset version - this avoided having to pass the version around internally as #5833 does for filemetadata. Now that that work is done, this code isn't needed. (While this make sure you can get filemetadatas from the file, you still don't know which one (and therefore had to pick draft or lastest public version). Sending the specific filemetadata or datasetversion let's you send the specific one being passed around).